dependent-sum/src/Data/GADT/Show.hs

```diff
@@ -22,9 +28,13 @@ import qualified Type.Reflection as TR
 -- like @(forall a. Show (t a)) => ...@.  The easiest way to create instances would probably be
 -- to write (or derive) an @instance Show (T a)@, and then simply say:
 --
--- > instance GShow t where gshowsPrec = showsPrec
+-- > instance GShow t
+--
+-- with any additional constraints as might be required by the Show instance.
 class GShow t where
     gshowsPrec :: Int -> t a -> ShowS
+    default gshowsPrec :: ForallF Show t => Int -> t a -> ShowS
+    gshowsPrec n (x :: t a) = whichever @Show @t @a (showsPrec n x)
```

Comment:
This should be upstreamed to `some`

Reply:
I think @Ericson2314 has another implementation there now, I'll confirm